From 61e899eeb3c9d6f2f37c10e7a879290355ec5a8d Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Thu, 9 Dec 2010 09:32:49 +0800
Subject: [PATCH 3/3] fsl_mpc8349e_mitx: adjust flash layout in dts

modify flash layout in dts for common use. Now the layout
is as follow:
dev:    size   erasesize  name
mtd0: 00020000 00010000 "hrcw"
mtd1: 000e0000 00010000 "dtb"
mtd2: 00600000 00010000 "kernel"
mtd3: 00100000 00010000 "u-boot-backup"
mtd4: 00020000 00010000 "hrcw-backup"
mtd5: 006e0000 00010000 "rootfs"
mtd6: 00100000 00010000 "u-boot"

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 arch/powerpc/boot/dts/mpc8349emitx.dts |   42 +++++++++++++++++++++++++------
 1 files changed, 34 insertions(+), 8 deletions(-)

diff --git a/arch/powerpc/boot/dts/mpc8349emitx.dts b/arch/powerpc/boot/dts/mpc8349emitx.dts
index bbb7c21..d79aeff 100644
--- a/arch/powerpc/boot/dts/mpc8349emitx.dts
+++ b/arch/powerpc/boot/dts/mpc8349emitx.dts
@@ -398,19 +398,45 @@
 			  0x3 0x0 0xf0000000 0x210>;	/* CF slot */
 
 		flash@0,0 {
-			compatible = "cfi-flash";
-			reg = <0x0      0x0 0x800000>;
-			bank-width = <2>;
-			device-width = <1>;
-		};
-
-		flash@0,800000 {
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "cfi-flash";
-			reg = <0x0 0x800000 0x800000>;
+			reg = <0x0 0x0 0x1000000>;
 			bank-width = <2>;
 			device-width = <1>;
+
+			hrcw@0 {
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			dtb@20000 {
+				reg = <0x20000 0xe0000>;
+			};
+
+			kernel@100000 {
+				reg = <0x100000 0x600000>;
+			};
+
+			u-boot-backup@700000 {
+				reg = <0x700000 0x100000>;
+				read-only;
+			};
+
+			hrcw-backup@800000 {
+				reg = <0x800000 0x20000>;
+				read-only;
+			};
+
+			rootfs@820000 {
+				reg = <0x820000 0x6e0000>;
+			};
+
+			u-boot@f00000 {
+				reg = <0xf00000 0x100000>;
+				read-only;
+			};
+
 		};
 
 		pata@3,0 {
-- 
1.6.5.2

